### DISCUSSION TITLE
IBM i Debug Server has not been started yet ...

### DISCUSSION THREAD

**Discussion by GatternigRud1** [codefori/vscode-ibmi#1946] [Q&A]
When executing action "Start Debugging ..." we receive error message: "Please run the STRDBGSVR command from a user profile with enough authority".

The debugging service was started with QSECOFR rights.
We have also granted the programmer QSECOFR rights. 

Nevertheless, we always receive the same error message. 
What could be the cause?

**Comment by sebjulliand** [comment]
We cannot know for sure since that message is thrown by IBM's `IBM i Debug` Extension.

But you can at least check if the service is _actually_ running with this query (it must return a result):
```sql
select * from qsys2.netstat_info where local_port_name = 'is-debug-ile'
```

If it is running and the Debug extension still complains about it not running, then reach out to IBM for support.

**Comment by mkwan01** [comment]
IBM i Debug Service requires that the host debug router QB5ROUTER is running, which is started by the STRDBGSVR command. The debug service does not try to run STRDBGSVR by itself, because their required authorities are different. Please logon from a green screen session and run "STRDBGSVR" and tried to debug from vscode again. Please report back if you still see the message after the QB5ROUTER is running.

**Reply by GatternigRud1** [reply]
I still see the message

**Reply by mkwan01** [reply]
Have you confirmed that the QB5ROUTER job is running on the IBM i machine? If it is running, please also check the owner user profile of the QB5ROUTER job, and confirm whether QB5ROUTER and the debug service were started under different user profiles.

**Comment by GatternigRud1** [comment]
Now it works! 
The cause was that a port on our firewall was not activated. Morris Guan, thank you for your support.
